remix-debug/src/debugger/debugger.js

```diff
@@ -111,8 +112,13 @@ Debugger.prototype.debugTx = function (tx, loadingCb) {
   })
 
   this.vmDebuggerLogic = new VmDebuggerLogic(this.debugger, tx, this.step_manager, this.debugger.traceManager, this.debugger.codeManager, this.debugger.solidityProxy, this.debugger.callTree)
+  this.vmDebuggerLogic.start()
 
   this.step_manager.event.register('stepChanged', this, function (stepIndex) {
+    if (!stepIndex) {
+      return self.event.trigger('endDebug')
```

Comment:
what is that for?

Reply:
it's to signal the debugging has ended due to the stepIndex no longer being valid